File: week 1 - /mario-more/mario.c
#include <cs50.h>
#include <stdio.h>

void print_row(int spaces, int length);

int main(void)
{
    // prompt user to input a suitable value
    int height;
    do
    {
        height = get_int("height of pyramid? ");
    }
    while (height < 1 || height > 8);

    // build pyramid of input height
    for (int i = 0; i < height; i++)
    {
        print_row(height - i, i + 1);
    }
}

// Defining print_row
void print_row(int spaces, int length)
{
    // producing correct number of " "
    for (int i = 0; i < spaces - 1; i++)
    {
        printf(" ");
    }
    // producing correct number of "#" left of gap
    for (int j = 0; j < length; j++)
    {
        printf("#");
    }
    // producing the 2 spacing gap
    printf("  ");
    
    // producing correct number of "#" right of gap
    for (int j = 0; j < length; j++)
    {
        printf("#");
    }
    printf("\n");
}

File: Week 6 - Python/sentimental-mario-more/mario.py
from cs50 import get_int


def main():
    height = 0
    while (height < 1 or height > 8):
        height = int(get_int("Height of pyramid? "))

    for i in range(height):
        print_row(height - i, i + 1)
        print()
        i = i + 1


def print_row(spaces, length):
    for i in range(spaces - 1):
        print(" ", end="")

    for j in range(length):
        print("#", end="")

    print("  ", end="")

    for k in range(length):
        print("#", end="")


main()
